src/CorrectPlayerMovePredictionPacket.php

```diff
@@ -52,20 +59,33 @@ public function getTick() : int{ return $this->tick; }
 
 	public function getPredictionType() : int{ return $this->predictionType; }
 
+	public function getVehicleRotation() : ?Vector2{ return $this->vehicleRotation; }
+
 	protected function decodePayload(PacketSerializer $in) : void{
+		$this->predictionType = $in->getByte();
 		$this->position = $in->getVector3();
 		$this->delta = $in->getVector3();
+		if($this->predictionType === self::PREDICTION_TYPE_VEHICLE) {
+			$this->vehicleRotation = new Vector2($in->getFloat(), $in->getFloat());
```

Comment:
Seb's IDA dump showed there's an optional float after this

Reply:
> Seb's IDA dump showed there's an optional float after this

Afaik thats not for the current Minecraft version but for the next one, so I did not include it yet